microbial_thermodynamics/cell_growth_ODE.py

```diff
@@ -94,24 +94,156 @@ def dr(tr: float, a: float, R: float) -> float:
     return udr
 
 
+def calculate_E(v: float, R: float, Q: float = 0.45, m: float = 1e8) -> float:
+    """Calculates enzyme copy number.
+
+    Args:
+     v: the ith species' proportional expression level for reaction alpha
+     R: Ribosome fraction
+     Q: Housekeeping proteome fraction
+     m: Cell mass
+    """
+    u = m * (v - R - Q)
+    return u
+
+
+def calculate_kappa(
+    reaction_energy: float,
+    Gatp: float = 75000,
+    G0: float = 1.5e5,
+    R: float = 8.314,
+    T: float = 293.15,
+) -> float:
+    """Calculates the kappa constant.
+
+    Args:
+     reaction_energy: the ATP generated for each species for each reaction alpha
+     Gatp: ATP free energy
+     G0: the standard Gibbs free-energy change when one mole of reaction alpha occurs
+     R: the gas constant
+     T: temperature
+    """
+    u = np.exp((((-G0) - reaction_energy) * Gatp) / R * T)
```

Comment:
Okay this equation is slightly wrong, as you multiple both `G0` and `reaction_energy` (eta in the paper) by `Gatp`. In fact it should only be `reaction_energy` multiplied by `Gatp`. Multiplying `G0` by `Gatp` results in a very large negative number which means that u goes to 0.

I'm pretty sure this is what the correct expression should be (though check it against the paper).
```python
u = np.exp(((-G0) - reaction_energy * Gatp) / R * T)
```